### DISCUSSION TITLE
Error with typeorm

### DISCUSSION THREAD

**Discussion by nickngqs** [medusajs/medusa#4239] [Q&A] [answered]
I keep getting this error when trying to deploy this on digitalocean app

I have updated my typeormm etc. 

I checked the upgrade docs and stuffs, but can't seem to get anything work. I managed to successfully build locally with no issues, is this possible there is something wrong with the migration script?

TypeError: typeorm_1.DataSource is not a constructor
[2023-06-02 10:57:04]     at /workspace/server/node_modules/@medusajs/medusa/dist/loaders/database.js:64:42
[2023-06-02 10:57:04]     at step (/workspace/server/node_modules/@medusajs/medusa/dist/loaders/database.js:33:23)
[2023-06-02 10:57:04]     at Object.next (/workspace/server/node_modules/@medusajs/medusa/dist/loaders/database.js:14:53)
[2023-06-02 10:57:04]     at /workspace/server/node_modules/@medusajs/medusa/dist/loaders/database.js:8:71
[2023-06-02 10:57:04]     at new Promise (<anonymous>)

**Comment by darwinva97** [comment]
same here, any update?

**Reply by rida-re** [reply]
Hello, 
You can fix that problem, try changing your dependancy  in package.json "typeorm" with this version : "^0.3.16".
you can use this command line: `yarn add typeorm`

**Comment by olivermrbl** [comment, marked as answer]
As @rida-rechrache noted, this is likely due to an issue with your Typeorm version. This error is thrown when the database is attempted to be initialized with version `0.2.*` of `typeorm` installed.

Can I get you to ensure you've updated your dependency, and that your deployment on DigitalOcean is not using some sort of cache? Unfortunately, I haven't deployed to DigitalOcean myself, so I cannot give you more specific actions to resolve your issue.

**Comment by nickngqs** [comment]
My typeorm is already in ^0.3.16.

I've also used digitalocean to force rebuild, clear cache and rebuild from scratch.

I might need to deep dive into digitalocean again. Is there a way to build locally to see what happen and serve locally?

**Reply by nickngqs** [reply]
@olivermrbl I think you're right, digitalocean might be caching the npm modules.

**Reply by olivermrbl** [reply]
Could you try creating a new DigitalOcean deployment to see if that changes anything?

**Reply by nickngqs** [reply]
I managed to deploy again, by deleting the npm modules and reinstalling. thanks for the help!

**Reply by olivermrbl** [reply]
Happy to hear!
